libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -675,7 +675,7 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				argumentTypes.push_back(arg->annotation().type);
 			}
 			utils().fetchFreeMemoryPointer();
-			utils().encodeToMemory(argumentTypes, TypePointers(), function.padArguments(), true);
```

Comment:
Please add `solAssert(!function.padArguments(), "");`

Reply:
It is moot I think. The only reason for `padArguments` to exists is to be used in `appendExternalFunctionCall`, though `SHA3` cannot be used in that context.

Reply:
Also it is used for a warning of inferring types. I'd actually rename it `usesPackedEncoding` or `usesTightPacking`.

Though that is the next step of refactoring making sense of this :)

Reply:
Still, at some point we might change the value and assume that this also changes the behaviour. Better more than fewer assertions.